Sources/NIO/BaseSocketChannel.swift

```diff
@@ -18,14 +18,16 @@ private struct SocketChannelLifecycleManager {
     // MARK: Types
     private enum State {
         case fresh
-        case registered
+        case lazilyRegistered // register() has been run but the selector doesn't know about it yet
+        case fullyRegistered // fully registered, ie. the selector knows about it
```

Comment:
Nit: I don't love these names. Can we call the new state `preRegistered`, maybe?